### PR TITLE
Build exporters based on new configuration

### PR DESCRIPTION
**Pull request by tigrannajaryan**
- Introduced a new unisvc binary and make target. Unisvc is
  planned to be the implementation of the unified agent and collector
  that uses the new configuration format.

- Refactored existing Application.execute() code to make it more
  readable and also reusable.
  The functionality is not changed. Reviews: please check this carefully!

- Introduced new Application.executeUnified() function that will call
  all new agent/collector execution logic that is different from
  old logic.
  Application.executeUnified() is currently partial implementation,
  which only builds the exporters. I plan to add building of processor
  pipelines and receivers in future PRs.

- Introduced ExportersBuilder which builds runtime exporters based
  on provided configuration.

- Added ability for App telemetry shutdown. This is required to be able to
  run multiple tests that involve App start and shutdown. Previously
  there was only one test - TestApplication_Start - and it was not
  correctly cleaning up telemetry resources at the end, making impossible
  to add more tests against App.

### CODE REVIEW

File: cmd/occollector/app/builder/exporters_builder.go [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+			// Remember that this data type is required for the exporter and also which
+			// pipeline the requirement is coming from.
+			result[exporter][pipeline.InputType] = dataTypeRequirement{pipeline}
+		}
+	}
+	return result
+}
+
+// Convert data type enum to string.
+func getDataTypeStr(dataType configmodels.DataType) string {
```
Comment (pjanotti) [inline cmd/occollector/app/builder/exporters_builder.go]: this private should be a public for configmodels.DataType.

File: cmd/occollector/app/builder/exporters_builder.go [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+
+	factory := factories.GetExporterFactory(config.Type())
+
+	exporter := &exporterImpl{}
+
+	inputDataTypes := exportersInputDataTypes[config]
+	if inputDataTypes == nil {
+		// No data types where requested for this exporter. This can only happen
+		// if there are no pipelines associated with the exporter.
+		eb.logger.Info("Exporter " + config.Name() +
```
Comment (pjanotti) [inline cmd/occollector/app/builder/exporters_builder.go]: nit: Warn instead of Info

File: cmd/occollector/app/builder/exporters_builder.go [hunk 3]
```diff
@@ -0,0 +1,252 @@
...
+}
+
+func typeMismatchErr(
+	config configmodels.Exporter,
+	requiredByPipeline *configmodels.Pipeline,
+	dataType configmodels.DataType,
+) error {
+	return fmt.Errorf("pipeline %q produces %q to exporter "+
+		"%s which does not support %q telemetry data. "+
+		"exporter will be detached from pipeline", requiredByPipeline.Name,
```
Comment (pjanotti) [inline cmd/occollector/app/builder/exporters_builder.go]: nit: `requiredByPipeline.Name` should be on the next line together with the other parameters 
(or each parameter on its own line, except format string)

**Comment by tigrannajaryan** [comment]
Everything fixed, please review again.

**Pull request merged**